### PR TITLE
Replace webpack with django_assets

### PR DESCRIPTION
**Pull request by Frost**
`django_assets` is a package that allows us to define our asset pipeline in python. No extra webpack needed.

### CODE REVIEW

File: assets/js/app.js [hunk 1]
```diff
@@ -21,4 +17,4 @@
     }
 
     document.addEventListener("readystatechange", bootstrap)
-})(window, document)
+})(window, document);
```
Comment (maxnordlund) [inline assets/js/app.js]:
```suggestion
})(window, document)
```

File: assets/js/booking_overview.js [hunk 2]
```diff
@@ -21,4 +21,6 @@ function setupBookingOverview() {
     )
 }
 
-exports.setupBookingOverview = setupBookingOverview
+(function(window) {
+    window.setupBookingOverview = setupBookingOverview
+})(window);
```
Comment (maxnordlund) [inline assets/js/booking_overview.js]:
```suggestion
})(window)
```

File: assets/js/payment.js [hunk 3]
```diff
@@ -87,4 +87,6 @@ function setupPayment(config) {
     cancelDiscountButton.addEventListener("click", closeDiscountForm)
 }
 
-exports.setupPayment = setupPayment
+(function(window) {
+    window.setupPayment = setupPayment
+})(window);
```
Comment (maxnordlund) [inline assets/js/payment.js]:
```suggestion
})(window)
```

File: assets/js/select_seats.js [hunk 4]
```diff
@@ -118,4 +118,6 @@ function setupSelectSeats(config) {
     )
 }
 
-exports.setupSelectSeats = setupSelectSeats
+(function(window) {
+    window.setupSelectSeats = setupSelectSeats
+})(window);
```
Comment (maxnordlund) [inline assets/js/select_seats.js]:
```suggestion
})(window)
```

**Comment by Frost** [comment]
@maxnordlund See [a83d62f](/Karspexet/Karspexet/pull/88/commits/a83d62fc84318404bb0ccd7487e250b375f72779) for another way of fixing your suggestions.

The reason I manually inserted those semicolons is because the bundling gets messed up when we have one file ending in `)`, and the next one starting with `"use strict"`. I also like defaults, so I would _really_ prefer the `semi: "always"` rule.

(regarding linters, I would also like to run `black` on the entire codebase, but that will definitely be left for another pull request).

**Comment by maxnordlund** [comment]
> @maxnordlund See [a83d62f](/Karspexet/Karspexet/pull/88/commits/a83d62fc84318404bb0ccd7487e250b375f72779) for another way of fixing your suggestions.

Darn it, you're too fast 😄 

> The reason I manually inserted those semicolons is because the bundling gets messed up when we have one file ending in `)`, and the next one starting with `"use strict"`.

That's... messed up. I though bundlers were supposed to handled that. Reminds me of the good old days, long before grunt, when `cat` was the only preprocessor. 

> I also like defaults, so I would _really_ prefer the `semi: "always"` rule.

Hm, defaults can be nice, but I _really_ dislike semi colons. Especially since you don't use them in Python and that context switch IMO just adds cognitive load. Is there no way to convince django-assets/webassets to splice in a semicolon before?

Otherwise you can use this IIFE form:

```js
!function(window, document) {
  // stuff
}(window, document)
```

The `!` typecast turns the function definition into an expression, allowing a direct call like that. 

> (regarding linters, I would also like to run `black` on the entire codebase, but that will definitely be left for another pull request).

I have to check that out as well, so many tools I didn't know about.

**Comment by maxnordlund** [comment]
I looked a bit closer and have two observations.

First, since looks something like:
```js
"use strict";

(function() {
  // ...
})()
```

There's already a semicolon before the IIFE, and concatenating them is safe since:

```js
... })()"use strict";(function() { ...
```

Doesn't parse as an extra call (which we're trying to avoid).

_But_ that also means everything, including third-party libraries, is running in strict mode. Not just our code. I seem to recall Yahoo trying this out and breaking production around the time strict mode came out.

**Comment by maxnordlund** [comment]
I also discovered that you can [write your own filters](https://webassets.readthedocs.io/en/latest/custom_filters.html#the-very-easy-way) in webassets.

```py
def noop(_in, out, **kw):
    out.write(_in.read())
```

Which could be used to do the wrapping automatically (which should prevent people leaking globals etc).

```py
def iife(_in, out, **kw):
    out.write(";(function(document, window) {\n")
    out.write('"use strict";\n')
    out.write(_in.read())
    out.write("})(document, window);\n")
```

File: assets/js/app.js [hunk 5]
```diff
@@ -21,4 +17,4 @@
     }
 
     document.addEventListener("readystatechange", bootstrap)
-})(window, document)
+}(window, document);
```
Comment (maxnordlund) [inline assets/js/app.js]:
```suggestion
}(window, document)
```

File: assets/js/booking_overview.js [hunk 6]
```diff
@@ -21,4 +21,6 @@ function setupBookingOverview() {
     )
 }
 
-exports.setupBookingOverview = setupBookingOverview
+!function(window) {
+    window.setupBookingOverview = setupBookingOverview
+}(window);
```
Comment (maxnordlund) [inline assets/js/booking_overview.js]:
```suggestion
}(window)
```

File: assets/js/payment.js [hunk 7]
```diff
@@ -87,4 +87,6 @@ function setupPayment(config) {
     cancelDiscountButton.addEventListener("click", closeDiscountForm)
 }
 
-exports.setupPayment = setupPayment
+!function(window) {
+    window.setupPayment = setupPayment
+}(window);
```
Comment (maxnordlund) [inline assets/js/payment.js]:
```suggestion
}(window)
```

File: assets/js/select_seats.js [hunk 8]
```diff
@@ -118,4 +118,6 @@ function setupSelectSeats(config) {
     )
 }
 
-exports.setupSelectSeats = setupSelectSeats
+!function(window) {
+    window.setupSelectSeats = setupSelectSeats
+}(window);
```
Comment (maxnordlund) [inline assets/js/select_seats.js]:
```suggestion
}(window)
```

**Comment by Frost** [comment]
The IIFE version still does not help.

One of the files starts with `"use strict"', which means that we end up with this in the bundle:

```
!function(window) { window.thingy = thingy }(window)"use strict"...
```

This is what's causing the syntax error, so it doesn't matter if we write it as `(function() {})()` or `!function(){}()`. It'll break anyway.

Your idea of writing a custom filter however, might be an idea.

Either that, or just assuming that `window` is a thing that exists and just polluting my the global namespace from the start.

**Comment by maxnordlund** [comment]
Hell yeah, let's go for that. No broken windows didn't hurt nobody. If you want you can try the filter thingy, but otherwise let it be.

**Comment by Frost** [comment]
Ugh. Now I have something that actually works, does _not_ have the semicolons in the js files, and has icon fonts working both when autogenerating the assets and when running them through collectstatic etc as we do in production.

@maxnordlund Care to have a look at my latest commits?

**Pull request merged**